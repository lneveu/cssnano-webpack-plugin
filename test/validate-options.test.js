import CssnanoWebpackPlugin from '../src';

it('validation', () => {
  /* eslint-disable no-new */
  expect(() => {
    new CssnanoWebpackPlugin({ test: /foo/ });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ test: 'foo' });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ test: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ test: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ test: ['foo', 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ test: [/foo/, 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ test: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssnanoWebpackPlugin({ test: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssnanoWebpackPlugin({ include: /foo/ });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ include: 'foo' });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ include: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ include: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ include: ['foo', 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ include: [/foo/, 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ include: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssnanoWebpackPlugin({ include: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssnanoWebpackPlugin({ exclude: /foo/ });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ exclude: 'foo' });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ exclude: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ exclude: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ exclude: ['foo', 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ exclude: [/foo/, 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ exclude: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssnanoWebpackPlugin({ exclude: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssnanoWebpackPlugin({ sourceMap: true });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ sourceMap: false });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ sourceMap: { inline: true } });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ sourceMap: 'true' });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssnanoWebpackPlugin({ cssnanoOptions: {} });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ cssnanoOptions: null });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssnanoWebpackPlugin({
      cssnanoOptions: { colormin: true },
    });
  }).not.toThrow();

  expect(() => {
    new CssnanoWebpackPlugin({ unknown: true });
  }).toThrowErrorMatchingSnapshot();
  /* eslint-enable no-new */
});
