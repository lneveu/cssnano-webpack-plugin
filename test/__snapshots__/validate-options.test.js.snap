// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options.test should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   Details:
    * options.test should be an array:
      [RegExp | non-empty string, ...]
    * options.test should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.test should be an instance of RegExp.
       * options.test should be a non-empty string."
`;

exports[`validation 2`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options.test should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   Details:
    * options.test[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.test[0] should be an instance of RegExp.
       * options.test[0] should be a non-empty string."
`;

exports[`validation 3`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options.include should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   Details:
    * options.include should be an array:
      [RegExp | non-empty string, ...]
    * options.include should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.include should be an instance of RegExp.
       * options.include should be a non-empty string."
`;

exports[`validation 4`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options.include should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   Details:
    * options.include[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.include[0] should be an instance of RegExp.
       * options.include[0] should be a non-empty string."
`;

exports[`validation 5`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options.exclude should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   Details:
    * options.exclude should be an array:
      [RegExp | non-empty string, ...]
    * options.exclude should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.exclude should be an instance of RegExp.
       * options.exclude should be a non-empty string."
`;

exports[`validation 6`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options.exclude should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   Details:
    * options.exclude[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.exclude[0] should be an instance of RegExp.
       * options.exclude[0] should be a non-empty string."
`;

exports[`validation 7`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options.sourceMap should be one of these:
   object { … } | boolean
   -> Enables/Disables generation of source maps.
   Details:
    * options.sourceMap should be an object:
      object { … }
      -> Options for source map.
    * options.sourceMap should be a boolean."
`;

exports[`validation 8`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options.cssnanoOptions should be an object:
   object { … }
   -> Options for \`cssnanoOptions\`."
`;

exports[`validation 9`] = `
"Invalid options object. Cssnano webpack plugin has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { test?, include?, exclude?, sourceMap?, cssnanoOptions? }"
`;
